db/migrate/20230525081252_update_created_manually_flag_on_customers.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class UpdateCreatedManuallyFlagOnCustomers < ActiveRecord::Migration[7.0]
+  class Customer < ApplicationRecord
+    has_many :orders, class_name: "Spree::Order"
+    acts_as_taggable
+  end
+
+  module Spree
+    class Order < ApplicationRecord
+      belongs_to :customer
+      self.table_name = 'spree_orders'
+    end
+  end
+
+  def change
+    # We want to set the created_manually flag to true for all customers that
+    #  1. have at least one tag
+    #  2. or don't have any orders
+    #  3. or we created more than 1 minute before the first order
+    Customer.where(id: customers_with_at_least_one_tag)
+      .or(Customer.where.not(id: customers_with_at_least_one_order))
+      .or(
+        Customer.where(id: customers_created_more_than_1min_before_orders)
+        .where.not(id: customers_created_no_more_than_1min_before_orders)
```

Comment:
@mkllnk  @dacook 

Here's what I was thinking about:

Suppose that we have a customer (c1) created at 00:05.

This customer has two orders: O1 (created at 00:05) and O2 (created at 00:07).

The first part of the query `customers_created_more_than_1min_before_orders` will return c1 because he has O2.
Now, we cannot suppose that this customer was created manually because of the O1.
I used the second query `customers_created_no_more_than_1min_before_orders` to exclude.

Reply:
Ah, thanks for explaining.

Reply:
This is now outdated. Resolving.